src/main/software/amazon/event/ruler/JsonRuleCompiler.java

```diff
@@ -608,25 +594,41 @@ private static Patterns processAnythingButListMatchExpression(JsonParser parser)
         return AnythingBut.anythingButMatch(values, hasNumber);
     }
 
-    private static Patterns processAnythingButEqualsIgnoreCaseListMatchExpression(JsonParser parser) throws JsonParseException {
+    private static Patterns processAnythingButValuesSetMatchExpression(JsonParser parser, MatchType matchType)
+            throws JsonParseException {
         JsonToken token;
         Set<String> values = new HashSet<>();
-        boolean hasNumber = false;
         try {
             while ((token = parser.nextToken()) != JsonToken.END_ARRAY) {
                 switch (token) {
                     case VALUE_STRING:
-                        values.add('"' + parser.getText() + '"');
+                        String text = parser.getText();
+                        if (matchType != MatchType.ANYTHING_BUT_IGNORE_CASE && text.isEmpty()) {
+                            barf(parser, "Null prefix/suffix not allowed");
+                        }
+                        values.add((matchType != MatchType.ANYTHING_BUT_SUFFIX ? "\"" : "") +
+                                   text +
+                                   (matchType != MatchType.ANYTHING_BUT_PREFIX ? "\"" : ""));
                         break;
                     default:
-                        barf(parser, "Inside anything-but/equals-ignore-case list, number|start|null|boolean is not supported.");
+                        if (matchType == MatchType.ANYTHING_BUT_IGNORE_CASE) {
+                            barf(parser, "Inside anything-but/equals-ignore-case list, number|start|null|boolean is not supported.");
```

Comment:
can be more helpful for debugging if we change this to `"Inside anything-but/equals-ignore-case, " + token +" is not supported"`

Reply:
Perhaps, but this would require changing outward-facing error messages for existing types, which I'd prefer not to do without a strong reason. Also, one benefit to the existing message is it says _why_ the token is not supported. If we just print the token there, it might not be obvious it is of the wrong type.